pika/adapters/blocking_connection.py

```diff
@@ -344,6 +342,8 @@ def _handle_read(self):
         keeps track of how many frames have been written since the last read.
 
         """
+        if not hasattr(self, '_read_poller'):
+            raise exceptions.ConnectionClosed()
```

Comment:
@wjps, this is not the right fix because the raised exception would not be consistent with the state of the connection - i.e., connection.is_closed would still evaluate to False after this. It's the same problem as with the `AttributeError` exception handler that this PR removed.

The issue is reported in #412 and I believe that PRs #533 and #542 together address this issue, while at the same time maintaining consistent connection state. I didn't remove the `AttributeError` block in those PRs, though.